test/Test_Evaluator.re

```diff
@@ -643,7 +643,61 @@ in fn("hello")|},
             ),
           );
 
-        probe_test({|"" : (a=String)|}, uexp);
+        probe_test({|PROBE("a") : (a=String)|}, uexp);
+      },
+    ),
+    test_case(
+      "Evaluate probe around inferred singleton labeled tuple in pattern",
+      `Quick,
+      () => {
+        let npp = expected_probe_pat(_, []);
+        let np = expected_probe(_, []);
+        let p = (p, es: list(Grammar.exp_term(unit))) =>
+          expected_probe_pat(
+            Probe(npp(p), {refs: []}),
+            List.map(Grammar.Annotated.empty, es),
+          );
+        let npt = (t): Grammar.typ_t(list(Grammar.exp_t(unit))) => {
+          term: t,
+          annotation: [],
+        };
+        let uexp =
+          np(
+            Let(
+              npp(
+                Cast(
+                  p(
+                    Var("x"),
+                    [
+                      Tuple([
+                        {
+                          term:
+                            TupLabel(
+                              {term: Label("l"), annotation: ()},
+                              {term: String("a"), annotation: ()},
+                            ),
+                          annotation: (),
+                        },
+                      ]),
+                    ],
+                  ),
```

Comment:
Yeah, this looks right. This is asserting that the tuple shows up in the probe. Test should be a lot clearer after https://github.com/hazelgrove/hazel/pull/1530/files